src/MongoDB.Driver.Core/Core/Configuration/ClusterBuilderExtensions.cs

```diff
@@ -148,13 +148,15 @@ public static ClusterBuilder ConfigureWithConnectionString(this ClusterBuilder b
             // Server
 
             // Cluster
+            builder = builder.ConfigureCluster(s => s.With(connectionMode: connectionString.Connect));
```

Comment:
This is a bug in the driver itself, not in the test framework.

This only really affects the test framework though, when High creates a cluster the connection mode was handled correctly.